### PR TITLE
Clean up Log.java.

### LINKED ISSUE #17: Change logger tag
Right now the logger tag is hard coded with 'sl4a' in it.
We should probably use a different name like 'snippet-lib'.

### PR DESCRIPTION
**Pull request by adorokhine**
* Remove unneeded methods. The UI methods are not useful because snippets
  propagate all exceptions to client side.
* Change tag from sl4a to MoblySnippetLib.

Fixes #17

---
This change is [Reviewable](https://reviewable.io/reviews/google/mobly-snippet-lib/48)

### CODE REVIEW

**Comment by xpconanfan** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 27 at r1](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Ke-kVjWe5ShW9Kxl6_L:-Ke-kVjWe5ShW9Kxl6_M:bmrmzax) ([raw file](https://github.com/google/mobly-snippet-lib/blob/36cfb9bc414554e3e27f15f4d8a8e18dc2b584ec/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L27)):*
> ```Java
>         String className = fullClassName.substring(fullClassName.lastIndexOf(".") + 1);
>         int lineNumber = stackTraceElements[4].getLineNumber();
>         return "MoblySnippetLib." + className + ":" + lineNumber;
> ```

Is there a good place to document this? We should probably let users know what kind of logger prefix to expect.
Maybe we need a user manual kind of doc for snippet lib beyond examples.
Doesn't have to be done in this PR.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48)*

**Comment by xpconanfan** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 19 at r1](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Ke-l-dY8BPeQ1SZrPAn:-Ke-l-dY8BPeQ1SZrPAo:b-bx4z0p) ([raw file](https://github.com/google/mobly-snippet-lib/blob/36cfb9bc414554e3e27f15f4d8a8e18dc2b584ec/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L19)):*
> ```Java
> package com.google.android.mobly.snippet.util;
> 
> public final class Log {
> ```

Should this be `final`?
Maybe we should allow users to inherit this logger and snap on their own tag without having to copy this class just to change the prefix string.
e.g. MBS would probably want a logger very much like this one.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48)*

**Comment by adorokhine** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 19 at r1](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Ke-l-dY8BPeQ1SZrPAn:-Ke1xVqdtoBXMSnRgr72:bf32j8t) ([raw file](https://github.com/google/mobly-snippet-lib/blob/36cfb9bc414554e3e27f15f4d8a8e18dc2b584ec/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L19)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

Should this be `final`?
Maybe we should allow users to inherit this logger and snap on their own tag without having to copy this class just to change the prefix string.
e.g. MBS would probably want a logger very much like this one.
</blockquote></details>

Currently all the methods in this class are static, so how would this "inheritance" work?

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 27 at r1](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Ke-kVjWe5ShW9Kxl6_L:-Ke1xbS_CZ3RZjocGKtz:bdg7hhz) ([raw file](https://github.com/google/mobly-snippet-lib/blob/36cfb9bc414554e3e27f15f4d8a8e18dc2b584ec/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L27)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

Is there a good place to document this? We should probably let users know what kind of logger prefix to expect.
Maybe we need a user manual kind of doc for snippet lib beyond examples.
Doesn't have to be done in this PR.
</blockquote></details>

If you want it in a comment somewhere LMK where and I will add it. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48)*

**Comment by xpconanfan** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 19 at r1](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Ke-l-dY8BPeQ1SZrPAn:-Ke20Za-l_FzN5iybY8Y:boqdvt4) ([raw file](https://github.com/google/mobly-snippet-lib/blob/36cfb9bc414554e3e27f15f4d8a8e18dc2b584ec/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L19)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

Currently all the methods in this class are static, so how would this "inheritance" work?
</blockquote></details>

Right, I shouldn't have said 'inherit'.
Basically, is it possible to reuse some of the code here?
E.g. can we avoid duplicating this class in MBS to achieve a similar behavior, but use a different tag from "MoblySnippetLib".

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48)*

**Comment by xpconanfan** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 27 at r1](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Ke-kVjWe5ShW9Kxl6_L:-Ke21366ISLhqdxhkUPN:b-s8cn7i) ([raw file](https://github.com/google/mobly-snippet-lib/blob/36cfb9bc414554e3e27f15f4d8a8e18dc2b584ec/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L27)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

If you want it in a comment somewhere LMK where and I will add it. 
</blockquote></details>

Maybe just add a log line in the ex1 and a comment saying what the log line looks like?
It's the easiest way without starting a completely new 'user manual' for now...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48)*

**Comment by adorokhine** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 19 at r1](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Ke-l-dY8BPeQ1SZrPAn:-Ke_goeDMe1kQcIWeCLE:bc4hdbw) ([raw file](https://github.com/google/mobly-snippet-lib/blob/36cfb9bc414554e3e27f15f4d8a8e18dc2b584ec/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L19)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

Right, I shouldn't have said 'inherit'.
Basically, is it possible to reuse some of the code here?
E.g. can we avoid duplicating this class in MBS to achieve a similar behavior, but use a different tag from "MoblySnippetLib".
</blockquote></details>

Open to suggestions :) I'm not sure how we'd reuse it without either using inheritance and making Log an object you have to instantiate, or requiring you to pass the name of the apk into every method like the android log does.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48)*

**Comment by xpconanfan** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 19 at r1](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Ke-l-dY8BPeQ1SZrPAn:-Ke_hASQ56wRRgGAhYz0:b-qkxyqt) ([raw file](https://github.com/google/mobly-snippet-lib/blob/36cfb9bc414554e3e27f15f4d8a8e18dc2b584ec/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L19)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

Open to suggestions :) I'm not sure how we'd reuse it without either using inheritance and making Log an object you have to instantiate, or requiring you to pass the name of the apk into every method like the android log does.
</blockquote></details>

If it's possible to do it through inheritance, sgtm.
If it's a lot more work, let's get this in and deal with it later.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48)*

**Comment by xpconanfan** [comment]
:lgtm:

---

Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-:-Ke_hZCxk4gJVUAHNnh7:bnfp4nl)*

**Comment by adorokhine** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 19 at r1](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Ke-l-dY8BPeQ1SZrPAn:-Ke_hmmON9bWuYL3yuev:bsr0ljb) ([raw file](https://github.com/google/mobly-snippet-lib/blob/36cfb9bc414554e3e27f15f4d8a8e18dc2b584ec/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L19)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

If it's possible to do it through inheritance, sgtm.
If it's a lot more work, let's get this in and deal with it later.
</blockquote></details>

Doing it by inheritance means to replace every static log.i and other call with constructing a new static logger object in class scope and calling logger.i on that. Is that what you want?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48)*

**Comment by xpconanfan** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 19 at r1](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Ke-l-dY8BPeQ1SZrPAn:-Ke_iFKhTHJm6onCuVUg:bhiwj49) ([raw file](https://github.com/google/mobly-snippet-lib/blob/36cfb9bc414554e3e27f15f4d8a8e18dc2b584ec/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L19)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

Doing it by inheritance means to replace every static log.i and other call with constructing a new static logger object in class scope and calling logger.i on that. Is that what you want?
</blockquote></details>

Can we make those methods not static and make Log a singleton?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48)*

**Comment by adorokhine** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 19 at r1](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Ke-l-dY8BPeQ1SZrPAn:-Ke_mObs9lRYfb-8do97:b-cor428) ([raw file](https://github.com/google/mobly-snippet-lib/blob/36cfb9bc414554e3e27f15f4d8a8e18dc2b584ec/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L19)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

Can we make those methods not static and make Log a singleton?
</blockquote></details>

How do you picture initializing the singleton for the first time? There's no user-controlled code until you get to the facades.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48)*

**Comment by xpconanfan** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 19 at r1](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Ke-l-dY8BPeQ1SZrPAn:-Ke_mukuUCxoye5V1TYg:b-v602ss) ([raw file](https://github.com/google/mobly-snippet-lib/blob/36cfb9bc414554e3e27f15f4d8a8e18dc2b584ec/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L19)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

How do you picture initializing the singleton for the first time? There's no user-controlled code until you get to the facades.
</blockquote></details>

Can't we initialize Log upon launching?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48)*

**Comment by adorokhine** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 19 at r1](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Ke-l-dY8BPeQ1SZrPAn:-Ke_pt62tFczinz8uKR_:bycb36z) ([raw file](https://github.com/google/mobly-snippet-lib/blob/36cfb9bc414554e3e27f15f4d8a8e18dc2b584ec/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L19)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

Can't we initialize Log upon launching?
</blockquote></details>

Sure, with what constant?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48)*

**Comment by xpconanfan** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 19 at r1](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Ke-l-dY8BPeQ1SZrPAn:-Ke_qM4f1ryF4hdSDubi:bz406uy) ([raw file](https://github.com/google/mobly-snippet-lib/blob/36cfb9bc414554e3e27f15f4d8a8e18dc2b584ec/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L19)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

Sure, with what constant?
</blockquote></details>

Sounds like the constant should be read from the manifest? Then MBS can supply a different constant via manifest?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48)*

**Comment by adorokhine** [comment]
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions.

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 19 at r1](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Ke-l-dY8BPeQ1SZrPAn:-KemHVuV8gJw96yEajbk:b-896fix) ([raw file](https://github.com/google/mobly-snippet-lib/blob/36cfb9bc414554e3e27f15f4d8a8e18dc2b584ec/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L19)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

Sounds like the constant should be read from the manifest? Then MBS can supply a different constant via manifest?
</blockquote></details>

Done.

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 27 at r1](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Ke-kVjWe5ShW9Kxl6_L:-KemHYmBLfWbOAVwSTG5:b3y8s00) ([raw file](https://github.com/google/mobly-snippet-lib/blob/36cfb9bc414554e3e27f15f4d8a8e18dc2b584ec/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L27)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

Maybe just add a log line in the ex1 and a comment saying what the log line looks like?
It's the easiest way without starting a completely new 'user manual' for now...
</blockquote></details>

Added the comment to AndroidManifest.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48)*

**Comment by xpconanfan** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 32 at r2](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Keo4GP7EcE1_QgY-Z2Y:-Keo4GP7EcE1_QgY-Z2Z:b-iwcyl0) ([raw file](https://github.com/google/mobly-snippet-lib/blob/52b605af44b27f5217817bf23293d8276db4736b/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L32)):*
> ```Java
>     public synchronized static void initLogTag(Context context) {
>         if (APK_LOG_TAG != null) {
>             throw new IllegalStateException("Logger is being re-initialized");
> ```

nit: maybe the msg should make it clear that logger should not be re-initialized instead of just saying a fact.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48)*

**Comment by xpconanfan** [comment]
:lgtm:

---

Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-:-Keo4SLwMqGHg-9uhLa9:bnfp4nl)*

**Comment by xpconanfan** [comment]
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48)*

**Comment by adorokhine** [comment]
Reviewed 1 of 1 files at r1, 2 of 3 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java, line 32 at r2](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-Keo4GP7EcE1_QgY-Z2Y:-Kepf0QMoTRYsYqGztO_:b-896fix) ([raw file](https://github.com/google/mobly-snippet-lib/blob/52b605af44b27f5217817bf23293d8276db4736b/third_party/sl4a/src/main/java/com/google/android/mobly/snippet/util/Log.java#L32)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

nit: maybe the msg should make it clear that logger should not be re-initialized instead of just saying a fact.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48)*

**Comment by xpconanfan** [comment]
:lgtm:

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-snippet-lib/48#-:-KepfKW9erGq1PVuCIMD:bnfp4nl)*
